`OpenRA.Utility X --png` fails with NullReferenceException

### Issue Summary
When following the [tutorial from the wiki](https://github.com/OpenRA/OpenRA/wiki/Utility#how-to-convert-an-icon-with-openra-utility) on how to extract a PNG with the Utility, the final step fails with an exception on a clean install.

#### System Information
- **Operating System:** Mac OS 10.13.6
- **.NET / Mono Version:** Mono 5.10.1.47
- **OpenRA Version:** 8b8a14e0b8f5ac7384133821812bf0c99aec1d1f
- **Mod:** -

#### Additional Information:
- Steps to reproduce: follow the first three steps in the [tutorial on the wiki](https://github.com/OpenRA/OpenRA/wiki/Utility#how-to-convert-an-icon-with-openra-utility).

- Logs
```
Unhandled Exception:
System.NullReferenceException: Object reference not set to an instance of an object
  at OpenRA.Graphics.Palette.AsSystemPalette (OpenRA.Graphics.IPalette palette) [0x00070] in <b5f72610442a4a0d91b2c8de25a08d33>:0
  at OpenRA.Mods.Common.UtilityCommands.ConvertSpriteToPngCommand.OpenRA.IUtilityCommand.Run (OpenRA.Utility utility, System.String[] args) [0x0014d] in <d5b77b84bf774f8e92497dcab4864a81>:0
  at OpenRA.Program.Main (System.String[] args) [0x002af] in <1623067990b4428ea97422d1518f3e41>:0
[ERROR] FATAL UNHANDLED EXCEPTION: System.NullReferenceException: Object reference not set to an instance of an object
  at OpenRA.Graphics.Palette.AsSystemPalette (OpenRA.Graphics.IPalette palette) [0x00070] in <b5f72610442a4a0d91b2c8de25a08d33>:0
  at OpenRA.Mods.Common.UtilityCommands.ConvertSpriteToPngCommand.OpenRA.IUtilityCommand.Run (OpenRA.Utility utility, System.String[] args) [0x0014d] in <d5b77b84bf774f8e92497dcab4864a81>:0
  at OpenRA.Program.Main (System.String[] args) [0x002af] in <1623067990b4428ea97422d1518f3e41>:0
```

--- a/OpenRA.Game/Graphics/Palette.cs
+++ b/OpenRA.Game/Graphics/Palette.cs
@@ -45,11 +45,6 @@ namespace OpenRA.Graphics
 			for (var i = 0; i < Size; i++)
 				pal.Entries[i] = palette.GetColor(i);
 
-			// hack around a mono bug -- the palette flags get set wrong.
-			if (Platform.CurrentPlatform != PlatformType.Windows)
-				typeof(ColorPalette).GetField("flags",
-					BindingFlags.Instance | BindingFlags.NonPublic).SetValue(pal, 1);
-
 			return pal;
 		}
 

